flink-runtime/src/main/java/org/apache/flink/runtime/state/CheckpointStreamFactory.java

```diff
@@ -89,6 +91,13 @@ default void reusePreviousStateHandle(Collection<? extends StreamStateHandle> pr
      * @return true if it can be reused.
      */
     default boolean couldReuseStateHandle(StreamStateHandle stateHandle) {
+
+        // By default, the CheckpointStreamFactory doesn't support snapshot-file-merging, so the
+        // SegmentFileStateHandle type of stateHandle can not be reused.
+        if (stateHandle instanceof SegmentFileStateHandle) {
+            return false;
+        }
```

Comment:
IIUC, the `FileMergingSnapshotManagerBase#couldReusePreviousStateHandle` should return `false` when it encounters an instance other than `SegmentFileStateHandle`. Right?

Reply:
Yes, you are right. I have resolved it.